design/pipeline-artifacts.md

```diff
@@ -0,0 +1,145 @@
+# Pipeline Artifacts YAML shortcut
+
+**Status: PM spec**
+
+Pipeline Artifacts are the new way to move files between jobs and stages in your pipeline. They are hosted in Azure Artifacts and will eventually entirely replace FCS "Build Artifacts". Because moving files between jobs and stages is a crucial part of most CI/CD workflows, and because Pipeline Artifacts are expected to be the default way to do so, this spec proposes a YAML shortcut syntax for publishing and downloading artifacts.
+
+In this document, `artifact` and `downloadArtifact` refer specifically to Pipeline Artifacts. Artifacts are distinct from `resources`, which are various kinds of inputs like `pipelines`, `containers`, `repositories`, and `packages`. Ashok is driving the YAML spec for these inputs, and he and I (amullans) are in sync.
```

Comment:
I like this. I'm open to (and would like) to offer a provider model syntax for different kinds of things, but I'd also like to keep Pipeline Artifacts pure and first-class.